webapp/app/js/services/buildapi.js

```diff
@@ -32,7 +32,7 @@ treeherder.factory('thBuildApi', [
     return {
         retriggerJob: function(repoName, requestId) {
 
-            $http({
+            return $http({
```

Comment:
tldr; Return promise so UI instead of model can decide what error message to show.